### ISSUE TITLE
session.set_media doesn't do anything

### ISSUE THREAD

**Issue by Mystro256** [emesene/emesene#650]
I was testing the music plugin to fix issues and whatnot, but session.set_media() does seem to do anything. I have tested it with the official client and multiple emesene sessions. There doesn't seems to be any acknowledgement of anything when I use session.set_media(STRING).

Is this a papyon glitch or emesene? or is this a yet to be implemented functionality?

I'm using papyon version 0.5.5-1ubuntu1.2

**Comment by Mystro256** [comment]
on a further note, when using the music function in the official client, I can't see the playing music in emesene.

**Comment by c10ud** [comment]
where is this set_media method?

**Comment by Mystro256** [comment]
It's in e3/base/session.py I believe

Edit: Just released the title of the issue had a typo, my bad, fixed

**Comment by arielj** [comment]
I can confirm that setting some media with the music plugin has no effect on contacts (they can't see what I'm listening neither with WLM or emesene)

**Comment by Mystro256** [comment]
The method is: (in emesene/emesene/e3/base/Session.py)

```
def set_media(self, message):
        '''set the message of the session'''
        self.add_action(Action.ACTION_SET_MEDIA, (message,))
```

I've looked around and I don't quite understand how it works to be honest, I'll have to leave it to someone who's has more experience in this than I. Although, given that viewing the "listening to" message doesn't even work with emesene, I would think this is a papyon issue, no? or at least how emesene handles papyon in regard to these messages.

**Comment by marianoguerra** [comment]
I read it really fast but it seems that papylib doesn't implement _handle_action_set_media

in e3.papylib.Worker

**Comment by Mystro256** [comment]
Ah ic, so it's not implemented in papyon yet?

Do you think it would be a good idea to use session.set_message for now?

**Comment by c10ud** [comment]
try 19ab2de42213e855cb1806f90278468d5ecf1dd2

**Comment by marianoguerra** [comment]
On Tue, Jun 21, 2011 at 4:21 PM, Mystro256
reply@reply.github.com
wrote:

> Ah ic, so it's not implemented in papyon yet?

it's not implemented in the papylib bridge, maybe it's supported on
papyon (that means that we don't pass the info from emesene to the
library)

> Do you think it would be a good idea to use session.set_message for now?

yes, you should use it and we (aka cloud ;) add the support in papylib
later (aka now)

**Comment by c10ud** [comment]
> > Do you think it would be a good idea to use session.set_message for now?
> > yes, you should use it and we (aka cloud ;) add the support in papylib
> > later (aka now)

don't, in papylib this triggers a content_roaming request and i don't think the server will be happy if you trigger this like, every 3 minutes or so.. try the commit i just pushed and hope for the best! (no music plugins here)

**Comment by Mystro256** [comment]
@c10ud

It kinda works. The format is: (MUSICNOTES) SONG - ARTIST

If I use the string "test something," I get this on the other end: (MUSICNOTES) t - e

So it shows the first letter in the string as the song and the second letter as the artist, lol?

**Comment by Mystro256** [comment]
There is also the issue of showing the media message on the receiving end. The test I did (above) was with emesene on the sending (using set_media) and WLM XP on virtualbox for the receiving end.

Also I noticed another issue/inconsistency. Calling "self.session.set_media(None)" on emesene sets the message back to the personal message on WLM's end but sets the message to blank on emesene's end (i.e, the music sender). I would think setting it back to the personal message would be the more desired effect. I tried set_message("test") afterwards and the message remained blank in the emesene window.

**Comment by c10ud** [comment]
about "self.session.set_media(None)": i think it could have something to do with self.session.media_change_succeed(message) in papylib/Worker.py.. try changing that line to
        if message is not None:
            self.session.media_change_succeed(message)

about music format, here's what papyon is doing:
        if current_media is not None:
            cm ='\0Music\01\0{0} - {1}\0%s\0%s\0\0' % \
                (xml_utils.escape(current_media[0]),
                 xml_utils.escape(current_media[1]))

so it's actually expecting a tuple for current media, not a string unfortunately..i guess the plugin needs a refactor because i don't see how we can hack papyon's implementation. it forces us to give [string0] - [string1] to the server..
as for your example, the music plugin instead of sending plain text you should send a tuple ("test", "something"), but then i need to fix:
        self.session.media_change_succeed(message)
        self.session.contacts.me.media = message
to:
        self.session.media_change_succeed(message.join(" - "))
        self.session.contacts.me.media = message.join(" - ")
or something similar

this also means the templates in music plugin cannot work

**Comment by Mystro256** [comment]
Well given the situation, the templates probably should be disabled, or at least for now. Unfortunately the music plugin seems somewhat useless and getting it to work is probably more important.

Your proposed solution seems like the best option, I would only recommend adding an "<i>(8)</i>" or some sort of notification to the user to show that it's a media message rather than their personal message being shown. I personally find adding the music note adds a nice touch :)
<img src=http://i55.tinypic.com/2sb0bpc.png>

Edit: Mind my bad grammar, I am on pain killers haha

**Comment by Mystro256** [comment]
Method set_media fixed by bc37468, further suggestions are implemented by #688
